modules/040-node-manager/hooks/handle_draining.go

```diff
@@ -189,6 +208,48 @@ func handleDraining(input *go_hook.HookInput, dc dependency.Container) error {
 	return nil
 }
 
+func getDrainTimeout(input *go_hook.HookInput, client k8s.Client, ngName string) time.Duration {
+	nodeGroups, err := client.Dynamic().Resource(nodeGroupResource).Namespace("").List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		input.Logger.Error("Failed to list node groups")
+		return defaultDrainTimeout
+	}
+
+	for _, group := range nodeGroups.Items {
+		groupName := group.GetName()
+		if groupName != ngName {
+			continue
+		}
+
+		spec, found, err := unstructured.NestedMap(group.Object, "spec")
+		if err != nil || !found {
+			input.Logger.Error("Error extracting 'spec' from node group", "ngName", groupName, "error", err)
+			input.Logger.Debug("NodeGroup object", "object", group.Object)
+			return defaultDrainTimeout
+		}
+
+		timeoutValue, found, err := unstructured.NestedFieldCopy(spec, "nodeDrainTimeout")
+		if err != nil || !found {
+			input.Logger.Error("Error extracting 'nodeDrainTimeout' from 'spec'", "ngName", groupName, "error", err)
+			input.Logger.Debug("Spec object", "spec", spec)
+			return defaultDrainTimeout
+		}
+
+		if intVal, ok := timeoutValue.(int64); ok {
```

Comment:
we do not understand wy do we need checking on float and int here?
Any way, marshaling to struct will left this checking

Reply:
changed to ngv1.NodeGroup